### ISSUE TITLE
ACL and http form upload

### ISSUE THREAD

**Issue by auguster** [andrewgaul/s3proxy#89]
I read in the s3proxy documentation that it is possible to 
> set and get canned bucket and object ACLs (private and public-read only).

Where can I find documentation about those ACL ?

I'm using the filesystem providers for testing purposes. I read on jclouds documentation that:
>By default, every item you put into a container is private, if you are interested in giving access to others, you will have to explicitly configure that. Exposing public containers is provider-specific.

The page about [filesystem provider](http://jclouds.apache.org/guides/filesystem/) doesn't mention ACL.

Basically, I'm trying to upload files from an HTML form and download, all directly to s3proxy using the official AWS documentation on [Creating an HTML Form](http://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-HTTPPOSTForms.html).

Whatever I attempt, I always get the following error:
```xml
<Error>
<Code>AccessDenied</Code>
<Message>
AWS authentication requires a valid Date or x-amz-date header
</Message>
<RequestId>4442587FB7D0A2F9</RequestId>
</Error>
```
Though x-amz-date is (correctly) set, I tried multiple variant of date. So I'm trying to make the bucket public to bypass the authentification issues.

**Comment by auguster** [comment]
I didn't see that s3proxy doesn't support POST upload.
The question still remains for http get and public-read acl.

**Comment by andrewgaul** [comment]
@auguster Did you test with S3Proxy 1.4.0 release or master?  The latter has several anonymous access mode fixes, specifically 618c35054c050ae2d92e9632e7987485de132096 and be3f38b937b62a13a2021a8510d7031a4bbd6ba1.  Generally S3Proxy only supports setting the private and public-read canned ACLs, either during object creation or during put object ACL.  Can you provide a minimized test case via some combination of the AWS CLI and curl so I can reproduce your symptoms?

**Comment by auguster** [comment]
I'm still using the 1.4.0 release, I haven't tried the one you gave me in #88 .
I finally understood what the canned ACLs were. For future reference [here is the documentation](http://docs.aws.amazon.com/cli/latest/reference/s3api/put-object.html).

I'll try to see if my understanding of the ACL and upgrading to future s3proxy solve my issues. If not, I'll try to give a minimum working example using curl.

**Comment by auguster** [comment]
I just deployed the 1.5.0 version you gave me, with the correct ACL, I can retrieve content from s3proxy without any trouble from the url of any webbrowser.

Making a HTML form with the correct AWS signature is a bit more complex. I'll need to figure this out. Does s3proxy support POST with `multipart/form-data` enctype ?

**Comment by andrewgaul** [comment]
#73 tracks the missing POST upload feature.  Once S3Proxy supports this, you would generate a signed URL which the HTML form submits via POST.  Closing this since master resolved the ACL issue and #73 tracks POST.
